pkgs/misc/emulators/ryujinx/default.nix

```diff
@@ -28,70 +15,40 @@ in stdenv.mkDerivation rec {
     sha256 = "00qql0wmlzs722s0igip3v0yjlqhc31jcr7nghwibcqrmx031azk";
   };
 
-  nativeBuildInputs = [ dotnet-sdk_5 dotnetPackages.Nuget cacert makeWrapper wrapGAppsHook gobject-introspection gdk-pixbuf ];
+  projectFile = "Ryujinx.sln";
+  executables = [ "Ryujinx" ];
+  nugetDeps = ./deps.nix;
 
-  nugetDeps = linkFarmFromDrvs "${pname}-nuget-deps" (import ./deps.nix {
-    fetchNuGet = { name, version, sha256 }: fetchurl {
-      name = "nuget-${name}-${version}.nupkg";
-      url = "https://www.nuget.org/api/v2/package/${name}/${version}";
-      inherit sha256;
-    };
-  });
+  nativeBuildInputs = [ wrapGAppsHook gobject-introspection gdk-pixbuf ];
+  buildInputs = [
+    gtk3
+    libX11
+    libgdiplus
+    ffmpeg
+    SDL2_mixer
+    openal
+    libsoundio
+    sndio
+    pulseaudio
+  ];
 
   patches = [
     ./log.patch # Without this, Ryujinx attempts to write logs to the nix store. This patch makes it write to "~/.config/Ryujinx/Logs" on Linux.
   ];
 
-  configurePhase = ''
-    runHook preConfigure
-
-    export HOME=$(mktemp -d)
-    export DOTNET_CLI_TELEMETRY_OPTOUT=1
-    export DOTNET_NOLOGO=1
-
-    nuget sources Add -Name nixos -Source "$PWD/nixos"
-    nuget init "$nugetDeps" "$PWD/nixos"
-
-    # FIXME: https://github.com/NuGet/Home/issues/4413
-    mkdir -p $HOME/.nuget/NuGet
-    cp $HOME/.config/NuGet/NuGet.Config $HOME/.nuget/NuGet
-
-    dotnet restore --source "$PWD/nixos" Ryujinx.sln
-
-    runHook postConfigure
-  '';
-
-  buildPhase = ''
-    runHook preBuild
-    dotnet build Ryujinx.sln \
-      --no-restore \
-      --configuration Release \
-      -p:Version=${version}
-    runHook postBuild
-  '';
-
-  installPhase = ''
-    runHook preInstall
-
-    dotnet publish Ryujinx.sln \
-      --no-build \
-      --configuration Release \
-      --no-self-contained \
```

Comment:
`--no-self-contained` is removed and so does `discordchatexporter-cli`. Does it matter? Would it cause dotnet runtime to be copied to $out which bloats output size?

If this option is required, maybe we need to change the default wrapper and set them in default `dotnetFlags`.

Reply:
> `--no-self-contained` is removed and so does `discordchatexporter-cli`. Does it matter? Would it cause dotnet runtime to be copied to $out which bloats output size?

I think you're right about the closure bloat, somehow completely missed the flag. I'll add it back to the default wrapper as you suggested :+1: